definition_validator/validator.cc

```diff
@@ -329,8 +329,9 @@ void validateOverriding(const core::Context ctx, core::SymbolRef method) {
         }
         auto isRBI = absl::c_any_of(method.data(ctx)->locs(), [&](auto &loc) { return loc.file().data(ctx).isRBI(); });
         if (!method.data(ctx)->isOverride() && method.data(ctx)->hasSig() &&
-            overridenMethod.data(ctx)->isOverridable() && !anyIsInterface && overridenMethod.data(ctx)->hasSig() &&
-            !method.data(ctx)->isRewriterSynthesized() && !isRBI) {
+            (overridenMethod.data(ctx)->isOverridable() || overridenMethod.data(ctx)->isOverride()) &&
```

Comment:
This conditional is getting pretty huge, but the meat of the change is that we raise this error if the parent is `overridable` or `override`, based on the assumption that the parent is correct.